sections/main-product.liquid

```diff
@@ -48,262 +52,42 @@
-                data-media-id="{{ section.id }}-{{ featured_media.id }}"
-              >
-                {%- assign media_position = 1 -%}
-                {% render 'product-thumbnail', media: featured_media, media_count: media_count, position: media_position, desktop_layout: section.settings.gallery_layout, mobile_layout: section.settings.mobile_thumbnails, loop: section.settings.enable_video_looping, modal_id: section.id, xr_button: true, media_width: media_width, lazy_load: false %}
-              </li>
-            {%- endif -%}
-            {%- for media in product.media -%}
-              {%- unless media.id == product.selected_or_first_available_variant.featured_media.id -%}
-                <li
-                  id="Slide-{{ section.id }}-{{ media.id }}"
-                  class="product__media-item grid__item slider__slide{% if single_media_visible %} product__media-item--single{% endif %}{% if product.selected_or_first_available_variant.featured_media == null and forloop.index == 1 %} is-active{% endif %}{% if media.media_type != 'image' %} product__media-item--full{% endif %}{% if section.settings.hide_variants and variant_images contains media.src %} product__media-item--variant{% endif %}"
-                  data-media-id="{{ section.id }}-{{ media.id }}"
-                >
-                  {%- liquid
-                    assign media_position = media_position | default: 0 | plus: 1
-                    assign lazy_load = false
-                    if media_position > 1
-                      assign lazy_load = true
-                    endif
-                    render 'product-thumbnail', media: media, media_count: media_count, position: media_position, desktop_layout: section.settings.gallery_layout, mobile_layout: section.settings.mobile_thumbnails, loop: section.settings.enable_video_looping, modal_id: section.id, xr_button: true, media_width: media_width, lazy_load: lazy_load
-                  -%}
+    <div class="grid__item product__media-wrapper{% if section.settings.media_position == 'right' %} medium-hide large-up-hide{% endif %} testBF">
+      <product-images-slider>
+        <div class="product-images__carousel glide" data-glide-product-images>
+          <div class="glide__track glide__track--product-images" data-glide-el="track">
+            <ul class="glide__slides glide__slides--product-images">
+              {% for image in product.images %}
+                <li class="glide__slide glide__slide--product-images">
+                  <div>
+                    <img src="{{ image | image_url: width: "450x" }}" loading="lazy" alt="{{ image.alt }}" class="image--product-slider">
```

Comment:
For image sizing, of the source image, it's reccomended they are 2 times what they will be displayed at.
The first big image that a user sees should not be lazyloaded, so you can use the forloop.first liquid object and property to test for the first image in this loop, and not lazyload that image.